nuxeo-compound-documents/src/test/java/org/nuxeo/compound/documents/TestCompoundDocument.java

```diff
@@ -16,65 +16,109 @@
 @Deploy("org.nuxeo.compound.documents")
-public class TestCompoundDocument extends AbstractJsonWriterTest.Local<DocumentModelJsonWriter, DocumentModel> {
+public class TestCompoundDocument {
 
     @Inject
-    protected CoreSession session;
-    
-    public TestCompoundDocument() {
-        super(DocumentModelJsonWriter.class, DocumentModel.class);
+    protected ServletContainerFeature servletContainerFeature;
+
+    protected HttpClientTestRule httpClientRule;
+
+    protected ObjectMapper mapper = new ObjectMapper();
+
+    protected Set<String> subtypes = Set.of("File", "Picture", "Video", "Audio", "CompoundDocumentFolder");
+
+    @Before
+    public void before() {
+        httpClientRule = new HttpClientTestRule.Builder().credentials("Administrator", "Administrator")
+                                                         .url(String.format("http://localhost:%d/api/v1",
+                                                                 servletContainerFeature.getPort()))
+                                                         .build();
+        httpClientRule.starting();
     }
 
+    @After
+    public void after() {
+        httpClientRule.finished();
+    }
+
+    // Avoid useless warning for not finding dclistener
+    @Deploy("org.nuxeo.compound.documents:OSGI-INF/disable-dclistener-contrib.xml")
     @Test
     public void testCompoundDocument() throws IOException {
-        DocumentModel doc = session.createDocument(session.createDocumentModel("/", "test", "CompoundDocument"));
-        assertTrue(doc.getFacets().contains("CompoundDocument"));
-        RenderingContext ctx = RenderingContext.CtxBuilder.enrichDoc("subtypes").get();
-        JsonAssert json = jsonAssert(doc, ctx);
-        json = json.has("contextParameters").isObject();
-        json.properties(1);
-        json = json.has("subtypes").isArray();
-        json.childrenContains("type", "File", "Picture", "Video", "Audio", "CompoundDocumentFolder");
+        var formData = new MultivaluedMapImpl();
+        formData.putSingle("doctype", "CompoundDocument");
+        formData.putSingle("name", "myCompoundDocument");
+        try (CloseableClientResponse response = httpClientRule.post("/path/", formData)) {
+            assertEquals(SC_SEE_OTHER, response.getStatus());
+        }
+        try (CloseableClientResponse response = httpClientRule.get(
+                "/path/myCompoundDocument?" + MarshallingConstants.EMBED_ENRICHERS + ".document=subtypes")) {
+            checkSubtypes(response, "CompoundDocument");
+        }
+        createSubTypes("myCompoundDocument");
+        try (CloseableClientResponse response = httpClientRule.get("/path/myCompoundDocument/myCompoundDocumentFolder?"
+                + MarshallingConstants.EMBED_ENRICHERS + ".document=subtypes")) {
+            checkSubtypes(response, "Folderish");
+        }
+        createSubTypes("myCompoundDocument/myCompoundDocumentFolder");
+    }
+
+    protected void createSubTypes(String docPath) {
```

Comment:
`createSubTypes` and `checkSubtypes` have different purposes whereas they share an equivalent name.
1. `createSubTypes` could be replaced by a method to create document and be renamed accordingly, thus the loop could be in the test method + use it for the first document creation
2. `checkSubtypes` could be renamed as it checks things than the subTypes (facet), an `assertCoumpoundResponse` could be interesting.